src/panels/developer-tools/template/developer-tools-template.ts

```diff
@@ -281,6 +285,10 @@ class HaPanelDevTemplate extends LitElement {
             max(16px, env(safe-area-inset-left));
         }
 
+        .content.horizontal {
+          --code-mirror-max-height: calc(100vh - 389px);
```

Comment:
Where does the `389px` come from? Seems arbitrary

Reply:
389px comes from
```
    --code-mirror-max-height: calc(100vh
                                - var(--header-height)
                                - var(--paper-font-body1_-_line-height) * 3 - 1em * 2
                                - max(16px,env(safe-area-inset-top)) * 2 - max(16px,env(safe-area-inset-bottom)) * 2
                                - var(--ha-card-border-width,1px) * 2
                                - 179px
                              );
```
where 179px =
height of paper-tabs	48
header of edit-pane	76
margin-top of card-content in edit-pane	-8
height of card-actions	47
padding-bottom of card-content in edit-pane	16

i.e. 389px is composed from css variables & hard-coded values.

Reply:
It would be better to write the formula as best you can here instead of just `389px`. There will still inevitably be hardcoded values but at least it would be somewhat dynamic and would still work if the `--header-height` is changed for example